### PR TITLE
Ev/module access control

### PR DESCRIPTION
**Pull request by eyalvo-matrix**
Hi

This is a discussion on the module permission access control 
I encountered the following problem
user eyal belongs to 2 ACL groups , both groups has all the ACL permissions except ADMIN
One of the groups does not have permissions to see a module through the 
MODULES-> management modules -> access control 
as shown in the pictures below 

What is expected 
=========================================
The user eyal will be able to see the link (Dosage list)

what happens 
=========================================
The user eyal is not  able to see the link  (Dosage list)

4_admin
3_eyal
2_eyal
1_eyal

### CODE REVIEW

**Comment by bradymiller** [comment]
The simplest thing to do here is to simply remove the deny code from zh_acl_check() function (since this can't support an explicit deny the way it is in the code or database) :)

So, would remove these lines from your file:
804
810
812
815
816
821
822
827
828
change 829 to an if
831
832

**Comment by eyalvo-matrix** [comment]
Ok
I committed the changes as you instructed :)

**Comment by bradymiller** [comment]
Is it testing well for you?

File: library/acl.inc [hunk 1]
```diff
@@ -801,35 +801,23 @@
...
                         group_settings.group_id IN (".$groupPlacemakers.") AND acl_sections.`section_identifier` = ? ";
-    $sql_group_acl_denied  = $sql_group_acl_base . " AND group_settings.allowed = '0'";
+
     $sql_group_acl_allowed = $sql_group_acl_base . " AND group_settings.allowed = '1'";
 
     // Complete the group queries sql binding array
     array_push($groups_sql_param, $section_identifier);
 
-        $count_user_denied      = 0;
-        $count_user_allowed     = 0;
```
Comment (bradymiller) [inline library/acl.inc]: oops, don't remove this one

File: library/acl.inc [hunk 2]
```diff
@@ -801,35 +801,23 @@
...
         $count_group_denied     = 0;
```
Comment (bradymiller) [inline library/acl.inc]: and remove this one

File: library/acl.inc (hunk 1 above)
Comment (eyalvo-matrix) [inline library/acl.inc]: ok

**Comment by eyalvo-matrix** [comment]
Ok , I made the changes and checked, and it is working well now

File: library/acl.inc (hunk 1 above)
Comment (eyalvo-matrix) [inline library/acl.inc]: There is no need for $sql_group_acl_denied
and I changed it to 
    $count_group_allowed    = 0;
    $count_user_allowed     = 0;

File: library/acl.inc (hunk 2 above)
Comment (eyalvo-matrix) [inline library/acl.inc]: ok

**Comment by eyalvo-matrix** [comment]
Hi I applied the additional fixes

**Comment by bradymiller** [comment]
Looks reasonable. Is it testing well for you?

**Comment by eyalvo-matrix** [comment]
Good week , 
Indeed the code seem to test well
I think it is ready to push into the code base 
Eyal

**Pull request merged**